src/model/exceptions.py

```diff
@@ -25,3 +25,13 @@ def __str__(self):
 class PartNotFound(Exception):
     """Thrown by the ICE client when a requested genotype part is not found"""
     pass
+
+
+class ModelNotFound(Exception):
```

Comment:
How about inheriting from [`urllib.error.HTTPError`](https://docs.python.org/3.6/library/urllib.error.html?highlight=urllib%20error#urllib.error.HTTPError)? Or even directly instantiate it, unless we need the distinction, of course.
```python
HTTPError(reason="Model not found", code=404)
```

Reply:
You cannot really instantiate an `HTTPError` like that, it requires further arguments like url and headers, so it seems a bit overkill to use that here.

I also feel that keeping an interface with explicit exception types here allows callers to be http agnostic, which admittedly might not benefit the resources module since it would be nice to simply forward the message and http status code generically, but to me it makes more sense as a general principle.

With that in mind, the http status code should be discarded when raising the exception, and then explicitly set by the resources module, presumably to the same code, but for somewhat different reasons. However, I do agree that at least passing on the exception message is better than duplicating it, so I'll implement that.